instance/gandi.py

```diff
@@ -42,10 +42,13 @@ class GandiAPI():
     Gandi API proxy object
     """
 
-    def __init__(self, api_url='https://rpc.gandi.net/xmlrpc/'):
+    def __init__(self, api_url='https://rpc.gandi.net/xmlrpc/', client=None):
         # This is a map of domain_name => zone_id key-value pairs.
-        self._zone_id_cache = {}
-        self.client = xmlrpc.client.ServerProxy(api_url)
+        self._zone_id_cache = None
```

Comment:
@pomegranited I don't think this is a problem.  Every configuration change requires a restart of the IM.  Gandi zone IDs can't change, so only adding and removing domains is relevant, and this isn't going to happen often.